scripts/globals/mobs.lua

```diff
@@ -37,4 +37,51 @@ function onMobDeathEx(mob, player, isKiller, isWeaponSkillKill)
             player:setVar("testingTime_crea_count",player:getVar("testingTime_crea_count") + 1);
         end
     end
-end;
+end
+
+-- is a lottery NM already spawned or primed to pop?
+function lotteryPrimed(phList)
+    local nm;
+    for k,v in pairs(phList) do
+        nm = GetMobByID(v);
+        if (nm ~= nil and (nm:isSpawned() or nm:getRespawnTime() ~= 0)) then
+            return true;
+        end
+    end
+    return false;
+end
+
+-- potential lottery placeholder was killed
+function phOnDespawn(ph,phList,chance,cooldown)
+    local phId = ph:getID();
+    for k,v in pairs(phList) do
+        if (k == phId) then
```

Comment:
if you're looking for a specific key, you should just do phList[phId] instead of looping

Reply:
fixed